SofascoreAcademy/SofascoreAcademy/Network/BaseApiClient.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  BaseApiClient.swift
+//  SofascoreAcademy
+//
+//  Created by Akademija on 01.05.2024..
+//
+
+import Foundation
+
+class BaseApiClient {
+    
+    func sendRequest(url: URL?) async -> Result<Data, ApiError> {
+        guard let url = url else {
+            return .failure(.wrongRequest)
+        }
+        
+        do {
+            let (data, response) = try await URLSession.shared.data(for: URLRequest(url: url))
+            
+            guard let response = response as? HTTPURLResponse else {
+                return .failure(.notResults)
+            }
+            
+            switch response.statusCode {
+            case 200...299:
+                return .success(data)
+            case 404:
+                return .failure(.notFound)
+            default:
+                return .failure(.serverError(code: response.statusCode))
+            }
+        } catch {
+            return .failure(.unknown)
+        }
+    }
+    
+    func sendRequestAndParse<T: Decodable>(url: URL?, responseModel: T.Type) async -> Result<T, ApiError> {
+        let result: Result<Data, ApiError> = await sendRequest(url: url)
+        
+        do {
+            let data: Data = try result.get()
+            guard let decodedResponse = try? JSONDecoder().decode(responseModel, from: data) else {
+                return .failure(.parsingError)
+            }
+            return .success(decodedResponse)
+        } catch let error {
+            return .failure(error as? ApiError ?? .unknown)
+        }
+    }
+}
+
+extension BaseApiClient {
+    
+    private func urlWithPath(_ path: String) -> URL? {
+        var components = URLComponents()
+        components.scheme = "https"
+        components.host = "academy-backend.sofascore.dev"
+        components.path = path
```

Comment:
Ovo je dobar put, ali domena ti može biti static let metoda izdvojena u BaseApiClientu.

Reply:
nije mi jasno zasto bih to izdvajala u static let samo sa domenom, kada se onda treba path posebno stavljati u svakom metodi posebno, umjesto da ga se samo ovako u jednoj funkciji postavi i funkcija odmah vraca url
mislim da ako samo domenu postavim kao let onda imam vise linija koda u svakoj funkciji gdje se radi url
moguce da sam krivo skuzila na koji nacin bih mogla domenu staviti kao let